lib/ansible/modules/remote_management/manageiq/manageiq_tenant.py

```diff
@@ -0,0 +1,447 @@
+
+        return dict(changed=True, msg=result['message'])
+
+    def edit_tenant(self, tenant, name, description):
+        """ Edit a manageiq tenant.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        resource = dict(name=name, description=description, use_config_for_attributes=False)
+
+        # check if we need to update ( compare_tenant is true is no difference found )
+        if self.compare_tenant(tenant, name, description):
+            return dict(
+                changed=False,
+                msg="tenant %s is not changed." % tenant['name'],
+                tenant=tenant['_data'])
+
+        # try to update tenant
+        try:
+            result = self.client.post(tenant['href'], action='edit', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="failed to update tenant %s: %s" % (tenant['name'], str(e)))
+
+        return dict(
+            changed=True,
+            msg="successfully updated the tenant with id %s" % (tenant['id']),
+            tenant=result)
+
+    def create_tenant(self, name, description, parent_id):
+        """ Creates the tenant in manageiq.
+
+        Returns:
+            the created tenant id, name, created_on timestamp,
+            updated_on timestamp.
+        """
+        # check for required arguments
+        for key, value in dict(name=name, description=description, parent_id=parent_id).items():
+            if value in (None, ''):
+                self.module.fail_json(msg="missing required argument: %s" % key)
+
+        url = '%s/tenants' % (self.api_url)
+
+        resource = {'name': name, 'description': description, 'parent': {'id': parent_id}}
+
+        try:
+            result = self.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="failed to create tenant %s: %s" % (name, str(e)))
+
+        return dict(
+            changed=True,
+            msg="successfully created tenant %s with id " % name,
+            tenant=result['results'])
+
+    def tenant_quota(self, tenant, quota_key):
+        """ Search for tenant quotas object by tenant and quota_key.
+        Returns:
+            the quotas for the tenant, or None if no tenant quotas were not found.
+        """
+
+        tenant_quotas = self.client.get("%s/quotas?expand=resources&filter[]=name=%s" % (tenant['href'], quota_key))
+
+        return tenant_quotas['resources']
+
+    def tenant_quotas(self, tenant):
+        """ Search for tenant quotas object by tenant.
+        Returns:
+            the quotas for the tenant, or None if no tenant quotas were not found.
+        """
+
+        tenant_quotas = self.client.get("%s/quotas?expand=resources" % (tenant['href']))
+
+        return tenant_quotas['resources']
+
+    def update_tenant_quotas(self, tenant, quotas):
+        """ Creates the tenant quotas in manageiq.
+
+        Returns:
+            result
```

Comment:
duh :-)  Something like ``dict with `msg` and `changed`.`` would be better

Reply:
I will check if the comments are correct 👍